### PR TITLE
refactor: cobra.sampling.sampling

### PR DESCRIPTION
**Pull request by synchon**
* [x] description of feature/fix
* [x] tests added/passed

This PR refactors `cobra.sampling.sampling` module to improve Python 3.6+ compatibility.

### CODE REVIEW

**Comment by synchon** [comment]
Rebased on latest `devel`.

File: src/cobra/sampling/sampling.py [hunk 1]
```diff
@@ -1,51 +1,58 @@
...
+) -> pd.DataFrame:
     """Sample valid flux distributions from a cobra model.
 
-    The function samples valid flux distributions from a cobra model.
-    Currently we support two methods:
+    Currently, two methods are supported:
 
     1. 'optgp' (default) which uses the OptGPSampler that supports parallel
-        sampling [1]_. Requires large numbers of samples to be performant
+        sampling. Requires large numbers of samples to be performant
```
Comment (Midnighter) [inline src/cobra/sampling/sampling.py:28]: I find this sentence contradictory. `n < 1000` is a small sample size to me. So is the recommendation to run OptGP when n > 1000 or to run achr when n < 1000?

**Review by Midnighter — APPROVED** [review]
I think the one sentence in the docstring can be improved but otherwise it looks good.

File: src/cobra/sampling/sampling.py (hunk 1 above)
Comment (synchon) [reply src/cobra/sampling/sampling.py:28]: I have missed to fix this information. If you look at `OptGPSampler` docs, it is recommended to use when `n > 1000`. I'll update it. Thanks!

**Comment by synchon** [comment]
The tests fail due to connection failure when performing `test_web` tests. Shouldn't be a problem for merging.

**Pull request merged**